tests/kafkatest/tests/streams/streams_application_upgrade_test.py

```diff
@@ -56,9 +55,9 @@ def perform_broker_upgrade(self, to_version):
             node.version = KafkaVersion(to_version)
             self.kafka.start_node(node)
 
-    @cluster(num_nodes=6)
-    @matrix(from_version=smoke_test_versions, bounce_type=["full"])
-    def test_app_upgrade(self, from_version, bounce_type):
+    @cluster(num_nodes=9)
+    @matrix(from_version=smoke_test_versions, bounce_type=["full"], metadata_quorum=[quorum.combined_kraft])
+    def test_app_upgrade(self, from_version, bounce_type, metadata_quorum):
```

Comment:
If `bounce_type` is always `full`, and `metadata_quorum` is always `quorum.combined_kraft`, why do we have both as test parameters? Can we not hard-code both inside the test itself?

Reply:
`metadata_quorum` must be in the `@matrix` annotation as it's picked up by the `ServiceQuorumInfo.for_test` class method to indicate the type of metadata quorum (via the `test_context`) - if it's not set, the test will assume you're using ZK and throws an error because the `zk` field in the `KafkaService` class is `None`

Reply:
IIRC (but not 100% sure) `full` means we stop all nodes first, and restart them afterward, while `rolling` would be regular rolling bounce, ie, stop-start, stop-start.

Not sure if we did run this test will rolling restart in the past? If yes, maybe there was a reason to remove rolling-restart and only test with `full`? Might need some digging. -- If it details this PR, we can also do a follow up to figure this out, and either add rolling-restart or maybe remove the parameter?

Reply:
From the history, it looks like the `rolling` parameter was never enabled; the code is there to support the `rolling restart.  So I enabled it in this latest run, and there were three failures, all from the `rolling` portion - it's from the three oldest versions.  If debugging these failures holds up the PR too much, I'll remove `rolling` and defer it to a follow up PR.

Reply:
So we keep as-is for now, and do a follow up PR to cleanup parameters is we don't need them or add other options?

Reply:
yes, I would advocate for this approach